modules/aws/s3_test.go

```diff
@@ -29,6 +30,37 @@ func TestCreateAndDestroyS3Bucket(t *testing.T) {
 	DeleteS3Bucket(t, region, s3BucketName)
 }
 
+func TestS3BucketTagging(t *testing.T) {
+	t.Parallel()
+
+	region := GetRandomStableRegion(t, nil, nil)
+	id := random.UniqueId()
+	logger.Logf(t, "Random values selected. Region = %s, Id = %s\n", region, id)
+
+	s3BucketName := "gruntwork-terratest-" + strings.ToLower(id)
+
+	CreateS3Bucket(t, region, s3BucketName)
+	defer DeleteS3Bucket(t, region, s3BucketName)
+
+	s3Client, err := NewS3ClientE(t, region)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tags := map[string]string{
+		"keyA": "valueA",
+		"keyB": "valueB",
+	}
+
+	addTagsToBucket(t, s3Client, region, s3BucketName, tags)
```

Comment:
Is this an async operation? Could checking tags immediately after adding them fail due to eventual consistency?

Reply:
Looks like it's a sync operation since the response is only returned when the operation is fulfilled.
AWS Go SDK has another function called `s3.PutBucketTaggingRequest`, and this one yes, is async, and requires polling to check the operation status